### PR TITLE
Replace the `bridge_utils` tool

### PR DESCRIPTION
**Pull request by luckyh**
Since not all distributions would support the `bridge_utils` tool,
we have to look for the alternative tools for better compatibility.

Signed-off-by: Xu Han <xuhan@redhat.com>

ID: 1622460

### CODE REVIEW

**Comment by luckyh** [comment]
@clebergnu @ldoktor @yashkmankad @kylazhang @chunfuwen @balamuruhans request for your review, thanks!

File: virttest/test_setup.py [hunk 1]
```diff
@@ -818,15 +818,15 @@ def _assemble_iptables_rules(self, port_list):
...
-        process.system("brctl addbr %s" % self.brname)
+        addbr_cmd = ("ip link add name %s type bridge stp_state 1"
+                     " forward_delay 400" % self.brname)
+        process.system(addbr_cmd)
         ip_fwd_path = "/proc/sys/net/%s/ip_forward" % self.ip_version
-        ip_fwd = open(ip_fwd_path, "w")
-        ip_fwd.write("1\n")
-        process.system("brctl stp %s on" % self.brname)
-        process.system("brctl setfd %s 4" % self.brname)
+        with open(ip_fwd_path, "w") as ip_fwd:
```
Comment (chunfuwen) [inline virttest/test_setup.py]: sound like original logic order is firstly change value in ip_fwd_path, then enable bridge.
After you change, it sound like you already enable bridge, then change value in ip_fwd_path.
 please confirm whether it impact behaviour somehow.

File: virttest/test_setup.py [hunk 2]
```diff
@@ -861,25 +861,24 @@ def _start_dhcp_server(self):
...
-        for line in output.split("\n"):
-            if line.startswith(self.brname):
-                # len == 4 means there is a TAP using the bridge
-                # so don't try to clean it up
-                if len(line.split()) < 4:
-                    return False
-        return True
+        # Return `True` when there is any TAP using the bridge
+        output = process.system_output("ip link show master %s" % self.brname)
+        output = decode_to_text(output).strip()
```
Comment (balamuruhans) [inline virttest/test_setup.py]: same here

**Comment by yashkmankad** [comment]
Reviewed-by: Yash Mankad <ymankad@redhat.com>

File: virttest/test_setup.py [hunk 3]
```diff
@@ -861,25 +861,24 @@ def _start_dhcp_server(self):
...
-        return decode_to_text(process.system_output("brctl show"))
-
     def _br_exist(self):
-        return self.brname in self._get_bridge_info()
+        result = process.run("ip link show %s type bridge" % self.brname,
+                             ignore_status=True)
+        if result.exit_status > 0:
+            return False
+        output = decode_to_text(result.stdout).strip()
+        if not output:
```
[the remark this answers is not in the document]
Comment (luckyh) [reply virttest/test_setup.py]: IIRC, people may run vt tests with using avocado 52.0 lts, so `decode_to_text()` will make them happy.

[the remark this answers is not in the document]
Comment (balamuruhans) [reply virttest/test_setup.py]: okay.

**Comment by balamuruhans** [comment]
Thanks all, let us get it in.

**Pull request merged**